src/thread_cache.h

```diff
@@ -213,29 +208,29 @@ class ThreadCache {
       ASSERT(list_ != NULL);
       length_--;
       if (length_ < lowater_) lowater_ = length_;
-      return SLL_Pop(&list_);
+      return FL_Pop(&list_);
     }
 
     bool TryPop(void **rv) {
-      if (SLL_TryPop(&list_, rv)) {
-        length_--;
-        if (PREDICT_FALSE(length_ < lowater_)) lowater_ = length_;
```

Comment:
We lost PREDICT_FALSE here

Reply:
PREDICT_FALSE can go inside Pop(). Or would you rather have the hint only in TryPop? If the latter, I will just replicate the length adjusting logic back in here.